### PR TITLE
fixed karmadactl promote

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin <wen.chen@daocloud.io>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
if there is not a namepasce of resource that be promoted in karmada control plane, it will throw a error: `namepace is not found`. 

So, before promote a resource, we should check whether the namespace is existing in control plane.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/karmadactl/promote.go [hunk 1]
```diff
@@ -278,6 +280,16 @@ func promote(controlPlaneRestConfig *rest.Config, obj *unstructured.Unstructured
 
 		fmt.Printf("Resource %q(%s) is promoted successfully\n", gvr, opts.name)
 	} else {
+		if _, err := controlPlaneClient.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{
```
[the remark this answers is not in the document]
Comment (wuyingjun-lucky) [reply pkg/karmadactl/promote.go:283]: How about add a flag like --createNs=true

[the remark this answers is not in the document]
Comment (lonelyCZ) [reply pkg/karmadactl/promote.go:283]: I think it is great. We can check if the namespace exists firstly.

[the remark this answers is not in the document]
Comment (calvin0327) [reply pkg/karmadactl/promote.go:283]: It's ok, if there is not a namespace that is same with the resource in control plane. we can prompt a warning to user.

[the remark this answers is not in the document]
Comment (wuyingjun-lucky) [reply pkg/karmadactl/promote.go:283]: Hi @calvin0327  Do you have any progress ? I encounter this problem and record in [1776](https://github.com/karmada-io/karmada/issues/1776).

[the remark this answers is not in the document]
Comment (calvin0327) [reply pkg/karmadactl/promote.go:283]: ok, I will close the pr

**Pull request closed without merging**